### DISCUSSION TITLE
How to remove old tls-san domain for api-server?

### DISCUSSION THREAD

**Discussion by shupcode** [k3s-io/k3s#8148] [Q&A] [answered]
Background:
I have a 3 master node HA k3s cluster.
In the setup phase i had an old domain added `tls-san=retired-domain.com`.

Goal:
Now I would like to migrate to a new domain name and remove the old tls-san `retired-domain.com`
I've tried removing the option and flag for all 3 master node service configs and restarting k3s but the cluster is still accepting connections from the old domain that I would like to retire. The old domain name is also appearing in the k8 secret `k3s-serving` annotations.

Is there an easy way to remove the old tls-san for security reasons when I move to a new domain?

**Comment by brandond** [comment, marked as answer]
See the thread at https://github.com/k3s-io/k3s/discussions/7420 regarding cleaning up the cache file and secret, which will cause the certificate to be reset to the default SANs.
